profiling/src/profile/mod.rs

```diff
@@ -58,6 +57,38 @@ impl From<&SampleId> for u64 {
 #[repr(transparent)]
 pub struct StackTraceId(usize);
 
+#[derive(Copy, Clone, Default, Debug, Eq, PartialEq, Hash)]
+#[repr(transparent)]
+pub struct LocationId(u32);
+
+impl LocationId {
+    pub fn new<T>(v: T) -> Self
+    where
+        T: TryInto<u32>,
+        T::Error: Debug,
+    {
+        let index = v
+            .try_into()
+            .expect("the machine to run out of memory far before this happens");
```

Comment:
I think using an error message like this is probably more descriptive and helpful:

```suggestion
            .expect("location ids to fit in u32");
```

Reply:
Done